File: python/2019/day_09.py
from pathlib import Path

from intcode import CPU


def main():
    # Read in the amp controller software
    text = (Path(__file__).parent / "../../input/2019/input_09.txt").read_text()
    program = [int(x) for x in text.split(',')]
    for p in [1, 2]:
        cpu = CPU(program)
        cpu.input.append(p)
        cpu.run()
        print(f"Part {p}:", cpu.output.popleft())


if __name__ == "__main__":
    main()

File: rust/2019/src/bin/day_09.rs
use std::fs;

use aoc_2019::CPU;

fn main() {
    let text = fs::read_to_string("../input/2019/input_09.txt").unwrap();
    let program: Vec<_> = text.trim().split(',').map(|s| s.parse().unwrap()).collect();
    for i in 1..=2 {
        let mut cpu = CPU::new(&program);
        cpu.input.push_back(i.into());
        cpu.run();
        println!("Part {}: {}", i, cpu.output.pop_front().unwrap());
    }
}
